src/main/java/com/google/sps/data/Chat.java

```diff
@@ -6,90 +6,133 @@
-      Chat c = new Chat(entity);
-      return c;
-    } catch (Exception e) {
-      return null;
+    private long id;
+    private List<String> participants;
+    private List<String> messages;
+
+    // Constructor 1
+    public Chat(long id, List<String> participants, List<String> messages) {
+        this.id = id;
+        this.participants = participants;
+        this.messages = messages;
+    }
+
+    // //Constructor 2
+    public Chat(Entity entity) {
+        this.id = (long) entity.getKey().getId();
+        this.participants = entity.getList("participants").stream().map(val -> ((StringValue) val).get())
+                .collect(Collectors.toList());
+        this.messages = entity.getList("messages").stream().map(val -> ((StringValue) val).get())
+                .collect(Collectors.toList());
+    }
+
+    public static Chat newChat(Datastore datastore, String participantId1, String participantId2) {
+        Chat chat = Chat.chatExists(datastore, participantId1, participantId2);
+
+        if (chat == null){
+            KeyFactory keyFactory = datastore.newKeyFactory().setKind("chat");
+            Key chatKey = datastore.allocateId(keyFactory.newKey());
+
+            Entity chatEntity = Entity.newBuilder(chatKey).set("participants", participantId1, participantId2)
+                    .set("messages", ListValue.of("Chat Start")).build();
+
+            datastore.add(chatEntity);
+            System.out.println("Breakpoint");
+            chat = new Chat(chatEntity);
+        }
+
+        return chat;
+    }
+
+    public static Chat getChatById(Datastore datastore, String chatId) {
+        Key chatKey = datastore.newKeyFactory().setKind("chat").newKey(Long.parseLong(chatId));
+        Entity chatEntity = datastore.get(chatKey);
+
+        Chat chat = new Chat(chatEntity);
+        System.out.println("Chat found");
+        return chat;
+    }
+
+    public static List<Chat> getChatsByUser(Datastore datastore, String participantId) {
+
+        List<Chat> chats = new ArrayList<Chat>();
+        Query<Entity> chatQuery = Query.newEntityQueryBuilder().setKind("chat")
+                .setFilter(CompositeFilter.and(PropertyFilter.eq("participants", participantId))).build();
+        QueryResults<Entity> chatEntities = datastore.run(chatQuery);
+
+        while (chatEntities.hasNext()) {
+            Entity entity = chatEntities.next();
+            Chat chat = new Chat(entity);
+            chats.add(chat);
+        }
+        return chats;
+    }
+
+    public static List<Chat> getAllChats(Datastore datastore) {
+
+        List<Chat> chats = new ArrayList<>();
+
+        Query<Entity> query = Query.newEntityQueryBuilder().setKind("chat").build();
+        QueryResults<Entity> chatEntities = datastore.run(query);
+
+        while (chatEntities.hasNext()) {
+            Entity entity = chatEntities.next();
+            Chat chat = new Chat(entity);
+            chats.add(chat);
+        }
+        return chats;
+    }
```

Comment:
These two could be refactored a bit more easily. You could use String `participantId` parameter and change the query based on whether it's null. Or even better, use a Java [optional](https://www.baeldung.com/java-optional) to not deal with null values:

```java
    // Gets all the chats, filtering on the given participant if present
    public static List<Chat> getAllChats(Datastore datastore, Optional<String> participantId) {
        EntityQuery.Builder query = Query.newEntityQueryBuilder().setKind("chat");
        if (participantId.isPresent()) {
          query = query.setFilter(PropertyFilter.eq("participants", participantId));
        }

        QueryResults<Entity> results = datastore.run(query.build());
        List<Chat> chats = new ArrayList<Chat>();
        results.forEachRemaining(entity -> chats.add(new Chat(entity)));
        return chats;
    }
```